redis/commands.cc

```diff
@@ -129,6 +129,26 @@ future<redis_message> echo::execute(service::storage_proxy&, redis::redis_option
     return redis_message::make_strings_result(std::move(_str));
 }
 
+shared_ptr<abstract_command> mset::prepare(service::storage_proxy& proxy, request&& req) {
+    std::map<bytes, bytes> data_map;
+    if (req.arguments_size() == 0 || req.arguments_size() % 2 != 0) {
+        throw wrong_number_of_arguments_exception(req._command);
+    }
+
+    for (unsigned i = 0; i < req.arguments_size() / 2; i++) {
+        data_map.insert(std::make_pair(req._args[2 * i], req._args[2 * i + 1]));
+    }
+
+    return seastar::make_shared<mset> (std::move(req._command), std::move(data_map), 0);
+}
+
+future<redis_message> mset::execute(service::storage_proxy& proxy, redis::redis_options& options, service_permit permit) {
+    return redis::write_strings_pairs(proxy, options, std::move(_data_map), _ttl, permit).then([] () {
+        return redis_message::ok();
```

Comment:
Contrary to redis, in Scylla such a write *can* fail. I don't know how to reconcile this with the required `ok` return status of the `mset` command.

Reply:
This is an interesting point!
If the write fails, I assume (I hope!) the write_strings_pairs() call will return an exceptional future, so hopefully the entire execution will return an error (which will hopefully - and this needs to be tested, at least manually - cause an error reply instead of the usual "OK". Which seems fine to me, but you're right that this isn't what the Redis documentation promises, and needs to be documented in redis.md - note that this is a bigger issue than just mset and the Redis promise isn't actually correct (see https://stackoverflow.com/questions/12911248/redis-set-command-cant-fail-but-can)